pep-XXXX.rst

```diff
@@ -0,0 +1,492 @@
+- The group that debated these proposals was split down the middle about whether these changes are even desirable:
+  - On the one hand they make callable types more expressive. On the other hand, they could easily confuse users who haven’t read the full specification of callable type syntax.
+  - We believe the simpler syntax proposed in this PEP, which introduces no new semantics and closely mimics syntax in other popular languages like Kotlin, Scala, and TypesScript, are much less likely to confuse users.
+- We intend to implement the current proposal in a way that is forward-compatible with the more complicated extended syntax. If the community decides after more experience and discussion that we want the additional features, they should be straightforward to propose in the future.
+- We realized that because of overloads, it is not possible to replace all need for Callback Protocols even with an extended syntax. This makes us prefer proposing a simple solution that handles most use cases well.
+
+We confirmed that the current proposal is forward-compatible with extended syntax by implementing a quick-and-dirty grammar and AST on top of the grammar and AST for the current proposal [#callable-type-syntax--extended]_.
+
+
+Syntax Closer to Function Signatures
+------------------------------------
+
+One alternative we had floated was a syntax much more similar to function signatures.
+
+In this proposal, the following types would have been equivalent::
+
+    class Function(typing.Protocol):
+        def f(self, x: int, /, y: float, *, z: bool = ..., **kwargs: str) -> bool:
+            ...
+
+    Function = (x: int, /, y: float, *, z: bool = ..., **kwargs: str) -> bool
+
+
+The benefits of this proposal would have included:
+- Perfect syntactic consistency between signatures and callable types.
+- Support for more features of function signatures (named, optional, variadic args) that this PEP does not support.
+
+Key downsides that led us to reject the idea include the following:
+- A large majority of use cases only use positional-only arguments, and this syntax would be more verbose for that use case, both because of requiring argument names and an explicit ``/``, for example ``(int, /) -> bool`` where our proposal allows ``(int) -> bool``
+- The requirement for explicit ``/`` for positional-only arguments has a high risk of causing frequent bugs - which often wouldn’t be detected by unit tests - where library authors would accidentally use types with named arguments.
+- Our analysis suggests that support for ``ParamSpec`` is key, but the scope rules laid out in PEP 612 would have made this difficult.
+
+
+Other Proposals Considered
+--------------------------
+
+An idea we looked at very early on was to allow using functions as types. This may be a great idea, but we consider less an alternative to better callable types than a major improvement in the usability of Callable Protocols:
```

Comment:
It's a little unclear what this "using functions as types" proposal is. Could you add a link to somewhere where it's fleshed out a little more? 🙂

Reply:
Yep! Unfortunately I don't think there's much to link to, everyone in the typing-sig discussion knows this from some meeings and emails but that's about it. I will see if there's a good email to link to, otherwise we may need to write a small doc on this.

Reply:
Clearly not "everyone", since I'm on typing-sig 😉

Reply:
(I think I joined about halfway through the discussion on improved callable syntax)

Reply:
Makes sense. It's an idea Łukasz Langa raised at the 2021 PyCon typing summit.

The idea is that we could allow writing something like
```
def MyCallbackProtocolType(x: int, /, y: str = ...): ...
```
in place of
```
class MyCallbackProtocolType(Protocol):
    def __call__(self, x: int, /, y: str = ...): ...
```
and then use the function itself as a type representing any callable with a compatible call signature.

Reply:
Ooh, that would be nice. But yes -- a great idea for a different PEP!

Reply:
I wrote up a doc on the idea because I don't want to inline too much about this proposal, since it mostly deserves a separate discussion not intermingled with comments about this PEP.
https://docs.google.com/document/d/1rv6CCDnmLIeDrYlXe-QcyT0xNPSYAuO1EBYjU3imU5s/edit

I linked to the doc, and also added a bit more detail directly in the PEP

Reply:
Ah, that's much better now! And yes, I agree that it's better to keep the details of that proposal out of the main body of this PEP :)

Reply:
A thought: perhaps this Google Doc could become an appendix to the PEP?